Scripts/DCS-BIOS/lib/modules/aircraft_modules/M-2000C.lua

```diff
@@ -131,65 +151,57 @@ local function segment_char_value(char)
 	return 0
 end
 
-local function add_pcn_segment_values(pcn_segment, segment_value, segment_index)
+local function add_pcn_segment_values(pcn_segment, segment_value, segment_index, digit_patterns, dot_patterns)
 	for index = 1, #segment_value do
-		local value = segment_value:sub(index, index)
+		local value = segment_char_value(segment_value:sub(index, index))
+
 		if not pcn_segment[index] then
 			pcn_segment[index] = {}
 		end
+		pcn_segment[index][segment_index] = value
 
-		pcn_segment[index][segment_index] = segment_char_value(value)
+		if segment_index == 7 then
+			dot_patterns[index] = (value > 0)
+		elseif value > 0 then
```

Comment:
Is this correct? Don't we want to store a dot pattern _and_ a digit for a case of like `.1`?

Reply:
Yes in my implementation that's correct, the idea is that dots information is separated from the digit information. In my head that make sit more flexible especially with other future displays in how we handle dots.

Reply:
oh sorry, I was being dumb, you're correct 👍